### ISSUE TITLE
Feature Requests

### ISSUE THREAD

**Issue by Vastrix** [fetacore/Infinitex#4]
Nice, a sexy Latex editor!  
Thank you, the world does indeed desperately need one :)

A few ideas, I'd like to run by you:

- **Latex Intellisense** (Would be really cool)
- The option write `$\latex$` in the *simple* editor (without pressing the button)
- Being able to quickly *export* any LaTeX as a transparent picture to the clipboard
  
  
Great program, regardless! :)

**Comment by fetacore** [comment]
Thank you very much for your kind words :D !
OK lets break it down so that I understand completely what you suggest:

- ### Latex Intellisense:
You mean more than just the suggestions that it already provides? There was a built in autocompletion option to get already written text as suggestions but [I scrapped it](https://github.com/fetacore/Infinitex/blob/4a611f32cc6a577bb3a6baeafca97028959942df/beforeBundle.js#L38) because it provided useless results that where accidentally selected on enter. It would be awesome to have an emmet implementation where, if you have a mistake that will prevent the document from compiling, an error mark would appear at the specific line left margin, but this will be a pain to implement and I have some more pressing issues in mind. If you are familiar with Ace Editor and Emmet you are welcome to implement it and submit a pull request! In case I am off-topic tell me.

- ### Latex in Simple
Well I tried to give native support in Quill for tex ($ and $$) but I was unsuccessful, so I [rewritten](https://github.com/fetacore/Infinitex/blob/4a611f32cc6a577bb3a6baeafca97028959942df/src/react/Editor.jsx#L27) the formula blot. I am not familiar with Parchment (the model Quill uses in the back) and I had a really hard time understanding it so I ended up with this solution. In general I use quill.deltas for the editor contents and not html and I could not change on $ press or on $$ press the content of the delta into a formula key with the insides as value. In theory it can be done but it requires knowledge of Parchment which I do not have.

- ### Export Latex as transparent picture
You mean with the math editor? Like an svg/png file or base64 string? You lost me here :P

**Comment by Vastrix** [comment]
Waw, amazing response! :)  
Lemme try to clarify some bits!
  
  
`You mean more than just the suggestions that it already provides?`
Right.. I'm a noob in LaTeX (but do find it very useful), so I was mainly testing the simple editor ;)  
But yes, like that but for math commands as well, not just for what seems to be markup commands(?)
![yea in this style](https://puu.sh/zC0d6/db9f64c9f2.gif)  
  
Yea, think you understood what I meant :)
Basically that it parses latex commands typed in the regular editor so you don't have to use the special input box ![](https://puu.sh/zC0kJ/a8f0597d35.png)  
  
  
Ah yea, lemme explain the third one :p  
See, I like mathematical formulas, however, sharing them, or just showing them to someone is a huge pain in the butt :s  
Lemme show you what I mean! :)  
![](https://puu.sh/zC0IG/dd5c2b32c1.gif)  
You would then have a picture on your clipboard of that `-1/12` equation  
In this example, it's not real LaTeX, but you get the point ;)  
I do understand that this request might be a bit off-topic.. :s  
  
  
Cheers o/

**Comment by fetacore** [comment]
OK so I managed to find a way to change the suggestions (but I could not add all of them). I also tried to make them somewhat intuitive, meaning that when you type \equation you get the snippet of \begin{equation} ... \end{equation} or some mathematical symbol snippets to appear like we call them, for example \muchgreaterthan translates to \gg which is >>. The work is not complete but if you or someone else have the time it would be really awesome to add more. On to your third request! For the second I really tried but quilljs is so tightly written that I cannot understand how to implement it (but I would really love to).
Check the new release and tell me what you think
Cheers!
